### DISCUSSION TITLE
Time to start cleaning up front-matter tags?

### DISCUSSION THREAD

**Discussion by hamishwillee** [mdn/mdn-community#294] [Content] [answered]
Tags in front-metadata are mostly very old legacy from wiki days, when they were used in a few places to generate dynamic lists. Only a small subset of tags are now used, and those are for formatting pages in the sidebar. 

IMO any that are useful should be moved to page metadata (this is already being planned for most cases) and the rest deleted.
What I'm hoping for at this point is broad agreement that it is worth deleting the ones that aren't used for anything now (and confirmation of what the set that are used is).

These tags position items in the sidebar, and will be replaced eventually by `page-type` metadata:
- `Reference`: indicates a page is API reference.
- `Method`: indicates a page is an API method
- `Property`: indicates a page is an API property
- Any others?

These tags indicate standardization level, and should be their own metadata:
- `Experimental` => `experimental: true`
- `Deprecated` => `deprecated: true`
- `Non-standard` => `non-standard: true`

I don't _think_ anything else is used for anything, even guide. 

Should start by getting rid of the guidance to add tags like "name of API" and also by stripping that out of the existing docs?

I'd like to delete anything that is not one of the above tags. So, for example:
  - Guide
  - HTML
  - Element
  - SVG
 - Glossary (and whatever the term is)
  - HTTP
  - HTTP Header
  - Request header
  - Response header
  - CSS
  - CSS Variables
  - Custom Properties
  - Web
  - cascading variables
  - CSS Background
  - CSS Property
  - recipe:css-property

**Comment by wbamberg** [comment]
As far as I can tell, there are all the places tags are used in KS: https://gist.github.com/wbamberg/1381c5c9c20752c89fed32b66c991230

I don't think they are used anywhere except KS.

- we could update CSSInfo and cssxref to use page types right now
- we could deal with "SVGRef" as part of https://github.com/mdn/yari/issues/7171
- if we defined page types for WebExtensions we could remove "Property", "Method" etc from WebExtAPISidebar, and that's the only place they are used
- if we defined page types for JS we could probably remove some of the usages in JSRef
- I expect we could scrap "Important" and "Junk" from SubpagesWithSummaries

....which leaves us really with just "Deprecated", "Obsolete", "Experimental", and "Non-standard".

**Reply by hamishwillee** [reply]
> ....which leaves us really with just "Deprecated", "Obsolete", "Experimental", and "Non-standard".

Yes. As I indicated at top, these could move to their own keys `deprecated: true` etc. (by default false). Except for obsolete, which MDN treats as deprecated (and so should move to `deprecated: true`)

We could update the KS to use these as well and do a mini project to add the keys.

> I expect we could scrap "Important" and "Junk" from SubpagesWithSummaries

Those aren't used. There are about 12 cases of SubpagesWithSummaries and none of them have those tags.

> For the rest ...

I think for the rest we just start creating those page types - once we have those, as you say, most of the rest goes.

I guess the question is whether we should start by deleting all the tags we know aren't "supported" now and change our guidance on how they are applied? That would immediately then drop the instructions down to at most: Add property or method, and experimental or deprecated or non standard, if any of these are true.

**Comment by Rumyra** [comment]
Sounds like it is about time @hamishwillee - and also I think this is a good example of opening a 'project' issue - I'm going to update the process this week, so I'll message you about how it's done (hopefully) before Friday :)

**Comment by hamishwillee** [comment, marked as answer]
This is being tracked in https://github.com/mdn/mdn/issues/262
